core/trino-main/src/main/java/io/trino/sql/gen/DereferenceCodeGenerator.java

```diff
@@ -44,7 +45,7 @@ public DereferenceCodeGenerator(SpecialForm specialForm)
         returnType = specialForm.getType();
         checkArgument(specialForm.getArguments().size() == 2);
         base = specialForm.getArguments().get(0);
-        index = (int) ((ConstantExpression) specialForm.getArguments().get(1)).getValue();
+        index = toIntExact((long) ((ConstantExpression) specialForm.getArguments().get(1)).getValue());
```

Comment:
Unrelated change?

Reply:
it's related

Reply:
How is it related? It's semantically equivalent (both are casts to int), so how did this fix anything?

Reply:
getValue() is now object of type Long. It cannot be cast to primitive int, without casting to primitive long first.